### PR TITLE
Ensure writes to UDP port happen after we start listening.

### PR DESCRIPTION
**Pull request by dkleckler**

### CODE REVIEW

File: src/test/java/com/flightstats/hub/metrics/StatsdReporterIntegrationTest.java [hunk 1]
```diff
@@ -42,15 +43,20 @@ private static StatsdReporter provideStatsDHandlers() {
         return provider.get();
     }
 
+    private static void runAfterSocketInit(Runnable runnable) {
+        Executors.newScheduledThreadPool(1).schedule(runnable, 100, TimeUnit.MILLISECONDS);
```
Comment (chriskessel) [inline src/test/java/com/flightstats/hub/metrics/StatsdReporterIntegrationTest.java]: I'm surprised this works. The returned Executor is local to the the function scope and I'd think it'd get garbage collected and thus stop rescheduling. Maybe sufficiently reliable in a unit test though.

**Comment by crozierm** [comment]
I'd be tempted to remove the static cancer from this class. I'll admit I didn't look at it very closely though.

File: src/test/java/com/flightstats/hub/metrics/StatsdReporterIntegrationTest.java (hunk 1 above)
Comment (dkleckler) [reply src/test/java/com/flightstats/hub/metrics/StatsdReporterIntegrationTest.java]: ScheduledThreadPoolExecutor has an onShutdown() hook that ensures delayed tasks execute even after shutdown (by default).  Periodic tasks will be canceled though.

**Comment by Paul-Hess** [comment]
@BeforeClass must be static, I find it useful, particularly here

**Comment by dkleckler** [comment]
Laurie killed off all the timed waits here, which should result in an optimally fast test execution.  Closing this one to PR over from her branch.

**Pull request closed without merging**